### ISSUE TITLE
Can't override attributes on task_policy on 2.4

### ISSUE THREAD

**Issue by mariotaddeucci** [apache/airflow#26534]
### Apache Airflow version

2.4.0

### What happened

Task policies on airflow 2.4 are raising ` ERROR [airflow.models.dagbag.DagBag] can't set attribute ` error on trying to override the values.

### How to reproduce

Override an attribute using task_policy
on config/airflow_local_settings.py using the docker image apache/airflow:slim-2.4.0-python3.10
```
def task_policy(task):
    task.owner = "newowner"
    task.on_failure_callback = lambda context: print(context)

```

### Operating System

docker image apache/airflow:slim-2.4.0-python3.10

### Deployment

Other Docker-based deployment

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by mariotaddeucci** [comment]
Works good with docker image apache/airflow:slim-2.4.0-python3.8

**Comment by mariotaddeucci** [comment]
Yes, it's happing on 3.9. On 3.8 only with dynamic task.
It happens with both parameters, owners and on failure callback.

**Comment by potiuk** [comment]
Any news here @mariotaddeucci ? Is this problem still present?

**Comment by dima-lyublin** [comment]
I have a similar issue in Airflow 2.4.2:
```
def create_hooks(self):
     self.s3_hook = S3Hook(aws_conn_id=self.s3_conn_id)
     self.s3_hook.extra_args = {"ContentType": content_type}
```
```
ERROR - Task failed with exception
Traceback (most recent call last):
  File "", line 57, in execute
    self.s3_hook.extra_args = {"ContentType": content_type}
AttributeError: can't set attribute
```

**Comment by potiuk** [comment]
@dima-lyublin Please do not hijack this discussion- your problem is that you are trying to override property. DON'T do it. _extra_args is the field that is exposed by the property.

**Comment by potiuk** [comment]
@mariotaddeucci -> any news ?

**Comment by potiuk** [comment]
Duplicate of #24547. Fixed in #28313
